src/tools/plan-generator.ts

```diff
@@ -94,31 +115,28 @@ export class PlanGenerator {
   }
 
   /** Generates a Kotlin `core.arc.type.Type` from a Type. */
-  // TODO(alxr): Implement
-  createType(type: Type): string {
+  async createType(type: Type): Promise<string> {
     switch (type.tag) {
       case 'Collection':
-        break;
+        return ktUtils.applyFun('CollectionType', [await this.createType(type.getContainedType())]);
```

Comment:
Totally optional: you could accept varargs for the function arguments, then you wouldn't have to create the list

Reply:
I'll consider it. Piotr had a suggestion to take in an `options` dictionary to fill all the arguments. I think it'd be good to couple the varargs and options changes together. I'd rather do this at a later time.